docs/proposals/integrations_release_cycle.md

```diff
@@ -0,0 +1,63 @@
+# Integrations release cycle
+
+- Authors: Massimiliano Pippi
+- Date: 2018-12-12
+- Status: draft
+- [Discussion](https://github.com/DataDog/integrations-core/pull/2741)
+
+## Overview
+
+Each agent based integration has its own version number and can be released as
+a Python package at any given time. This document outlines what the release
+cycle will be for each integration.
+
+## Problem
+
+At this moment the release cycle for most of the integration is not formalized
+and there's no process in place to keep it consistent and reliable. Once a feature
+or a bugfix is merged on `master`, different things may happen:
+
+- A release is immediately triggered.
+- A release is triggered after merging other, maybe unrelated changes for the same integration.
+- A release is triggered at a later time without any specific deadline.
+
+This causes a number of issues for all actors involved:
+
+- Engineers have to improvise on a case by case basis.
+- Contributors have no idea if changes within a PR will be released and when.
+- Users don't know when a feature or a bugfix will be actually available.
+
+## Constraints
+
+1. Each integration must keep its own release cycle.
+2. The solutions must work for integrations from `integrations-extra` as well.
+
+## Recommended Solution
+
+A release must be immediately triggered after merging a PR into the `master` branch.
+To ensure quality of the software shipped, **changes on the branch should be tested before merging**.
+A pre-release testing checklist is out of the scope of this RFC but in general
+an engineer must consider that changes in a PR could be used right after the
+release by any users on any supported Agent version and platform.
```

Comment:
Whats the recommended path for situations where we have multiple PRs to merge for a single release? Merge them all into a branch and then merge that  branch into master? Is the release automatic or is a human involved?

Reply:
Release is human involved and will always be since the commit that fires the build pipeline has to be signed by one of the authorized release managers.

Multiple PRs can be still merged on `master` and released at once, this RFC covers the release management process we should use going forward but exceptions will still be possible to cover situations like the recent changes to `kafka_consumer`, see @jeremy-lq comment here: https://github.com/DataDog/integrations-core/pull/2741#issuecomment-447067762.

Going to update the text with these info.